src/java.net.http/share/classes/jdk/internal/net/http/common/DebugLogger.java

```diff
@@ -53,60 +115,65 @@ final class DebugLogger implements Logger {
     private static final DebugLogger NO_HPACK_LOGGER =
-            new DebugLogger(HTTP, "HPACK"::toString, Level.OFF, Level.OFF);
+            new DebugLogger(HTTP, "HPACK"::toString, LoggerConfig.OFF);
     static final long START_NANOS = System.nanoTime();
 
     private final Supplier<String> dbgTag;
-    private final Level errLevel;
-    private final Level outLevel;
+    private final LoggerConfig config;
+    private final int minSeverity;
     private final System.Logger logger;
     private final boolean debugOn;
     private final boolean traceOn;
 
     /**
      * Create a logger for debug traces.The logger should only be used
      * with levels whose severity is {@code <= DEBUG}.
-     *
-     * By default, this logger will forward all messages logged to the supplied
-     * {@code logger}.
-     * But in addition, if the message severity level is {@code >=} to
-     * the provided {@code errLevel} it will print the messages on System.err,
-     * and if the message severity level is {@code >=} to
-     * the provided {@code outLevel} it will also print the messages on System.out.
+     * <p>
+     * By default, this logger will print message whose severity is
+     * above the severity configured in the logger {@code config}
+     * <ul>
+     *     <li>If {@code config.outLevel()} is not Level.OFF, messages
+     *     whose severity are at or above that severity will be directly
+     *     printed on System.out</li>
+     *     <li>If {@code config.errLevel()} is not Level.OFF, messages
+     *     whose severity are at or above that severity will be directly
+     *     printed on System.err</li>
+     *     <li>If {@code config.logLevel()} is not Level.OFF, messages
+     *     whose severity are at or above that severity will be forwarded
+     *     to the supplied {@code logger}.</li>
+     * </ul>
      * <p>
      * The logger will add some decoration to the printed message, in the form of
      * {@code <Level>:[<thread-name>] [<elapsed-time>] <dbgTag>: <formatted message>}
      *
      * @apiNote To obtain a logger that will always print things on stderr in
      *          addition to forwarding to the internal logger, use
-     *          {@code new DebugLogger(logger, this::dbgTag, Level.OFF, Level.ALL);}.
+     *          {@code new DebugLogger(logger, this::dbgTag,
+     *                                 LoggerConfig.LOG.withErrLevel(Level.ALL));}.
      *          To obtain a logger that will only forward to the internal logger,
-     *          use {@code new DebugLogger(logger, this::dbgTag, Level.OFF, Level.OFF);}.
+     *          use {@code new DebugLogger(logger, this::dbgTag, LoggerConfig.LOG);}.
      *
      * @param logger The internal logger to which messages will be forwarded.
-     *               This should be either {@link #HPACK} or {@link #HTTP};
+     *               This should be either {@link #WS}, {@link #HPACK}, or {@link #HTTP};
```

Comment:
Given that the constructor of `DebugLogger` is private and we only expose APIs to create `HTTP`, `WS` and `HPACK` loggers, this statement that the loggers be either of these is true. However, the constructor isn't doesn't do any checks for the `logger` instance (other than null check). Do you think we can drop this sentence?

Reply:
Reworded: `This is typically either ... `